### ISSUE TITLE
Also disable Sonar when TSLint rule is disabled inline

### ISSUE THREAD

**Issue by xxluke** [SonarSource/SonarTS#824]
**I want to request a feature.**
We use [tslint-sonarts](https://www.npmjs.com/package/tslint-sonarts) on the project before we check it with Sonar. When there is an issue we won't fix, we disable the lint rule by adding a `tslint:disable` comment. But then it's still active in Sonar, even it's the same rule:
![grafik](https://user-images.githubusercontent.com/8194761/51673000-35a0d400-1fcd-11e9-99b4-d15bc16bd23a.png)

Of course I could also add a `// nosonar`, but it's extra work which seems unneccessary and it's annoying since we don't notice the problem until we perform the whole analysis (we can't use the IDE plugin).
 
This might be a missing feature of Sonar in general, not limited to TypeScript.

**Comment by vilchik-elena** [comment]
@xxluke I don't think we want to consider tslint disable comments in SonarQube
In your use-case, if you really want to use TSLint, I would suggest to have an empty profile in SonarQube for TS and import tslint report. That way you will see only tslint issues left after `tslint:disable`. See docs on this feature [here](https://docs.sonarqube.org/display/PLUG/Importing+TSLint+issues).

**Comment by morris** [comment]
The bigger issue here is that `// nosonar` ignores all errors on a line, e.g.

```
if (4 === 4) { // nosonar
```

will not report any issues, as opposed to

```
if (4 === 4) { // tslint:disable-line:no-magic-numbers
```

where the no-identical-expressions will still be reported.

@vilchik-elena that would make SonarQube obsolete for us (we could just tslint on pre-push instead), or am I missing something? In any case, the feature described by @xxluke would be extremely helpful IMHO :)

**Comment by vilchik-elena** [comment]
@marris The idea behind SonarQube is that you mark issue as "won't fix" or "FP" in the SQ UI, so that you are not bothered by it in future. Usage of `NOSONAR` is not encouraged, so I don't think that any soon we will evolve it to support rule key.
SonarQube can bring much more other value compared to TSLint (metrics, quality gate, leak period, history, coverage, branches, many languages etc) but if you believe that for your project the main interest is to have 0 issues, I think indeed TSLint will be a more appropriate.

**Comment by morris** [comment]
> SonarQube can bring much more other value compared to TSLint (metrics, quality gate, leak period, history, coverage, branches, many languages etc) but if you believe that for your project the main interest is to have 0 issues, I think indeed TSLint will be a more appropriate.

Definitely, that's why we are using it in the first place :) we haven't really been using the UI workflow that much yet, but if that's an encouraged workflow it could work for us.

I still believe the requested feature is an enhancement worth looking at from SQ's perspective. I don't see any downside right now, and it would certainly make SQ's analysis more accurate.

**Comment by melgish** [comment]
// Disabled rule because the third-party API chokes on anything else.
// tslint:disable-next-line:any-rule-you-want
const runsWithScisors: any = 3;

Would it be that much more difficult to support both?  Have an option in SQ to ignore (the default) or respect the tslint:disable comments in source.  This way everyone is happy.  The existence of 'won't-fix' means accept that there will be times when rules will be broken out of necessity.  However a 'won't-fix' in sonar is out-of-sight, out-of-mind when the developer is in the code. However the above comment

// An extra comment because sonar doesn't count anything before first line of source ;-)
